### PR TITLE
AWS Docs

### PR DESCRIPTION
**Pull request by datosh**
Signed-off-by: Fabian Kammel <fk@edgeless.systems>

### Proposed change(s)

- Install docs
- Getting started

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

File: docs/docs/getting-started/first-steps.md
Comment (datosh) [inline docs/docs/getting-started/first-steps.md:172]: @malt3 could you help me out with that point? `config generate` does not fill that value in. I see that this is not a static value as with GCP or Azure, should we add some logic to CLI to fetch this value based on the region? Is a manual approach currently preferred?

Comment (datosh) [inline docs/docs/getting-started/first-steps.md:172]: @malt3 Is [this](https://github.com/edgelesssys/constellation/blob/feat/docs/aws-install/.github/workflows/build-os-image.yml#L270-L271) the full list of regions we target for release?

File: docs/docs/getting-started/first-steps.md
Comment (3u13r) [reply docs/docs/getting-started/first-steps.md:180]: Until https://github.com/edgelesssys/constellation/pull/416 is decided on and implemented, I think we maybe just add the mapping from region to ami as a comment behind the field by hand.

File: docs/docs/getting-started/first-steps.md
Comment (3u13r) [reply docs/docs/getting-started/first-steps.md:172]: Yes, we decided on having one image in every major region. Of course images from us-east-2 cannot be used in us-east-1, but I hope our users are flexible here. Supporting more location will be a pain, since we probably want to clean up old images at some point and we would have to do that in every region then.

Comment (malt3) [reply docs/docs/getting-started/first-steps.md:172]: We should add a disclaimer here (and maybe in the code if we see no AMI for the region) that users can open an issue to get the image replicated into their region.

Comment (malt3) [reply docs/docs/getting-started/first-steps.md:172]: I can create an issue template that we can link to for that purpose

File: docs/docs/getting-started/first-steps.md
Comment (malt3) [inline docs/docs/getting-started/first-steps.md:211]: The upload to S3 has to happen manually for now using each AMI ID. The download should work (I think). This will be fully supported by #416

File: docs/docs/workflows/scale.md
Comment (malt3) [inline docs/docs/workflows/scale.md:69]: No yet supported. See AB#2441

File: docs/docs/workflows/storage.md
Comment (malt3) [inline docs/docs/workflows/storage.md:42]: @daniel-weisse should know more about this :)

File: docs/docs/workflows/troubleshooting.md
Comment (malt3) [inline docs/docs/workflows/troubleshooting.md:41]: No

Comment (datosh) [reply docs/docs/workflows/troubleshooting.md]: Was implemented, added missing permissions in #461 .

File: docs/docs/getting-started/first-steps.md
Comment (datosh) [reply docs/docs/getting-started/first-steps.md]: I re-used the feature template with a prefilled title, feel free to improve this as you see fit :)

File: docs/docs/workflows/storage.md
Comment (datosh) [reply docs/docs/workflows/storage.md]: Added disclaimer.

File: docs/docs/workflows/storage.md
Comment (thomasten) [inline docs/docs/workflows/storage.md:46]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/workflows/recovery.md
Comment (daniel-weisse) [inline docs/docs/workflows/recovery.md:95]: Is there actually anything blocking us from supporting recovery?
As far as I know, this should just require us to set up a loadbalancer the CLI talks tom the rest should be CSP independent.

File: docs/docs/getting-started/install.md
Comment (Nirusu) [inline docs/docs/getting-started/install.md:265]: In my opinion the whole sentence is a bit awkward to read.
"Follow Amazon's guide on understanding and managing policies" by itself sounds fine, though I think both articles are actually about understanding the purpose of certain kinds of policies.

"Follow Amazon's guide on understanding and managing policies_, and roles_" sounds off to me. The comma implies a break in speech, yet "and roles" is meant to be read as "Follow Amazon's guide on understanding [...] roles" or "Follow Amazon's guide on managing [...] roles".

Not sure which one, though. And having two "ands" in the sentence is also not good, which makes it a confusing enumeration.

I would rewrite this, though I don't have a concrete suggestion as I don't fully know the articles and context of the linked articles to make a suggestion.

File: docs/docs/workflows/scale.md
Comment (Nirusu) [inline docs/docs/workflows/scale.md:105]: Why do we want users to create issues for everything we likely should support in the first place (and likely will add later anyways)? Or do we plan on abandoning AWS after the release of 2.2.0 until we get user feedback?

File: docs/docs/workflows/storage.md
Comment (Nirusu) [inline docs/docs/workflows/storage.md:44]: Same here, why do we suggest users to create an issue?

File: docs/docs/workflows/scale.md
Comment (Nirusu) [reply docs/docs/workflows/scale.md:105]: Ah, bad wording on my side, but I see.

I wasn't so sure about whether it was just not available since AWS was just recently implemented and therefore not everything is ready yet or if we actually deprioritize it / don't want to work on it for now as technically AWS also offers the weakest security model of the CSPs we support. 

An alternative suggestion I have is to reword this as _will become available in a later release_ but still suggest opening an issue if this is a priority for someone. I think that is more neutral and reflects that AWS is more in an early state.

But don't need to change it then if we actually don't want to focus on AWS for now until we know users actually want to use it. I think that's fine then.

File: docs/docs/getting-started/install.md
Comment (datosh) [reply docs/docs/getting-started/install.md:138]: I added more context to help with the motivation, @thomasten.

File: docs/docs/workflows/recovery.md
Comment (datosh) [reply docs/docs/workflows/recovery.md:95]: @3u13r is working on the loadbalancer part. I can take another stab at recovery once it's merged.

File: docs/docs/workflows/scale.md
Comment (datosh) [reply docs/docs/workflows/scale.md:105]: I hinted at the fact that we have those tasks in our backlog: "Scaling isn't **yet** implemented for AWS.", users should engage with us (via issues) if this is an urgent need.

I will resolve this conversation for now, feel free to reopen if you feel we should change/improve the wording.

File: docs/docs/getting-started/install.md
Comment (datosh) [reply docs/docs/getting-started/install.md]: I agree! I have removed the last link since it was off-context. The first two resources should provide enough information to know how policies work.

**Comment by datosh** [comment]
Wait for #416 to be merged, and update AMI-related information before merging.

File: docs/docs/workflows/recovery.md
Comment (daniel-weisse) [reply docs/docs/workflows/recovery.md:95]: Tested this with with a single control-plane cluster, and a 3 control-plane nodes cluster.
Both self recovery (at least half of the control-planes are still reachable) and manually recovery using the CLI work on AWS.

Comment (3u13r) [reply docs/docs/workflows/recovery.md:95]: We still need to manually pass the IP to `constellation recover`, right? So I see two ways forward:
1. Keep saying that `constellation recover` is not supported
2. Document the way of getting the right IP from AWS and recovering with that. 

I have no real opinion on which way we choose.

File: docs/docs/getting-started/first-steps.md
Comment (thomasten) [inline docs/docs/getting-started/first-steps.md:210]:
```suggestion
    * **iamProfileControlPlane**: The name of an IAM instance profile attached to all control-plane nodes.

        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. The profile name will be provided as Terraform output value: `control_plane_instance_profile`.

        Alternatively, you can create the AWS profile with a tool of your choice. Use the JSON policy in [main.tf](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam/main.tf) in the resource `aws_iam_policy.control_plane_policy`.

    * **iamProfileWorkerNodes**: The name of an IAM instance profile attached to all worker nodes.

        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. The profile name will be provided as Terraform output value: `worker_nodes_instance_profile`.
```

File: docs/docs/getting-started/install.md
Comment (thomasten) [inline docs/docs/getting-started/install.md:138]:
```suggestion
To set up a Constellation cluster, you need to perform two tasks that require permissions: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure.
```

File: docs/docs/overview/clouds.md
Comment (thomasten) [inline docs/docs/overview/clouds.md:35]:
```suggestion
AWS currently doesn't offer CVMs. AWS proprietary Nitro Enclaves offer some related features but [are explicitly not designed to keep AWS itself out](https://aws.amazon.com/blogs/security/confidential-computing-an-aws-perspective/). Besides, they aren't suitable for running entire Kubernetes nodes inside them. Therefore, Constellation uses regular EC2 instances on AWS [Nitro](https://aws.amazon.com/ec2/nitro/) without runtime encryption. Attestation is based on the [NitroTPM], which is a vTPM managed by the Nitro hypervisor. Hence, the hypervisor is currently part of Constellation's TCB.
```

File: docs/docs/workflows/storage.md
Comment (thomasten) [inline docs/docs/workflows/storage.md:131]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/workflows/storage.md
Comment (thomasten) [inline docs/docs/workflows/storage.md:299]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/getting-started/first-steps.md
Comment (3u13r) [inline docs/docs/getting-started/first-steps.md:204]: Since there is currently a [discussion on where to put the script](https://github.com/edgelesssys/constellation/pull/478), I'd suggest to pin a commit in the link. Even more so, because when we implement `constellation iam create` we will probably move those files again. 

I'm also fine with linking against the release branch, as the place _should_ be stable.

**Review by daniel-weisse — APPROVED** [review]

File: docs/docs/getting-started/install.md
Comment (3u13r) [inline docs/docs/getting-started/install.md:178]: `AmazonEC2FullAccess` alone is not sufficient since i.e. `logs:*` permissions form below are missing.

**Review by 3u13r — CHANGES_REQUESTED** [review]

**Review by 3u13r — APPROVED** [review]
LGTM

**Pull request merged**
